pdl-compiler/src/backends/rust/parser.rs

```diff
@@ -513,6 +533,108 @@ impl<'a> FieldParser<'a> {
                     }
                 });
             }
+            (ElementWidth::Dynamic(element_size_field), ArrayShape::Static(count)) => {
+                // The element width is known, and the array element
+                // count is known statically.
+                let array_size = if *count == 1 {
+                    quote!(#element_size_field)
+                } else {
+                    quote!(#count * #element_size_field)
+                };
+
+                self.check_size(&span, &array_size);
+
+                let parse_element =
+                    self.parse_array_element(&format_ident!("chunk"), width, type_id, decl);
+
+                self.code.push(quote! {
+                    // TODO: use
+                    // https://doc.rust-lang.org/std/array/fn.try_from_fn.html
+                    // when stabilized.
+                    let #id = #span.chunks(#element_size_field)
+                        .take(#count)
+                        .map(|mut chunk| #parse_element.and_then(|value| {
```

Comment:
Should we use `decode_full(chunk)` here ? or do you care about having a different error case ?

Reply:
We could use `decode_full` but that would mean complexifing the `parse_element` which I prefered to avoid and the different error case seems nice since `TrailingBytes` doesn't give you any information.